src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs

```diff
@@ -43,39 +42,30 @@ public class RecalcEngine : IScope, IPowerFxEngine
         /// <param name="powerFxConfig">Compiler customizations.</param>
         public RecalcEngine(PowerFxConfig powerFxConfig = null)
         {
-            _powerFxConfig = powerFxConfig ?? new PowerFxConfig();
-
-            AddFunction(BuiltinFunctionsCore.Index_UO);
-            AddFunction(BuiltinFunctionsCore.ParseJson);
-            AddFunction(BuiltinFunctionsCore.Table_UO);
-            AddFunction(BuiltinFunctionsCore.Text_UO);
-            AddFunction(BuiltinFunctionsCore.Value_UO);
-            AddFunction(BuiltinFunctionsCore.Boolean);
-            AddFunction(BuiltinFunctionsCore.Boolean_UO);
+            powerFxConfig = powerFxConfig ?? new PowerFxConfig(null, null);
+            AddInterpreterFunctions(powerFxConfig);
+            powerFxConfig.Lock();
+            _powerFxConfig = powerFxConfig;
         }
 
         // Add Builtin functions that aren't yet in the shared library. 
-        private void AddFunction(TexlFunction function)
+        private void AddInterpreterFunctions(PowerFxConfig powerFxConfig)
```

Comment:
>AddInterpreterFunctions [](http://example.com/codeflow?start=21&length=23)

Nit: 
Now that this is successfully decoupled from the interpreter...  could this be moved to the bottom of BuiltinFunctionsCore?  (next to where these functions are already defined)

Reply:
What I'd like is to  have BuiltinFunctionsCore contain a bunch of smaller sets of functions, and then each host would call like: AddUOFunctions, AddMathFunctions, AddStringFunctions,...that also take into account the Enums required for those functions. I think we'll want to address that as a separate PR, so I'll leave this here for now.